Matchmaker/Program.cs

```diff
@@ -1,119 +1,72 @@
 ﻿using System.Diagnostics;
 using Cassandra;
 
-var cassandra = new CassandraSetup(address: "127.0.0.1", port: 9043, replicationFactor: 3);
+var cassandraAddress = "127.0.0.1";
+var cassandraPort = 9042;
+var numberOfServersPerRegionAndGameType = 5;
+var numberOfPlayersPerRegionAndGameType = 30;
+var numberOfMatchmakers = 3;
+
+var cassandra = new CassandraSetup(cassandraAddress, cassandraPort, replicationFactor: 3);
 var mapper = cassandra.Mapper;
+Stopwatch stopwatch = new();
 
-#region create and save servers
+#region generate and start servers
 
-List<ServersSimulator> serversSimulators = new List<ServersSimulator>{
-    new ServersSimulator(30, 9042),
-    new ServersSimulator(30, 9043),
-    new ServersSimulator(30, 9044)
-};
+var serverGenerator = new ServerGenerator(cassandraAddress, cassandraPort);
+var statsCollector = new StatsCollector();
+stopwatch.Start();
+var servers = serverGenerator.Generate(numberOfServersPerRegionAndGameType, statsCollector);
+stopwatch.Stop();
+Console.WriteLine($"generated {servers.Count()} servers in time: {stopwatch.ElapsedMilliseconds} ms");
+var serversSimulator = new ServersSimulator();
+serversSimulator.SimulateServers(servers);
 
-List<Task> serverTasks = new();
+#endregion
 
-Stopwatch s1 = new();
-s1.Start();
+#region create and start matchmakers
 
-foreach (var simulator in serversSimulators)
+var matchmakerSimulators = Enumerable.Range(0, numberOfMatchmakers).Select(_ =>
 {
-    serverTasks.Add(simulator.SimulateServers());
-}
+    var simulator = new MatchmakerSimulator(cassandraAddress, cassandraPort);
+    var thread = new Thread(new ThreadStart(simulator.SimulateMatchmaker));
+    thread.Start();
+    return (thread, simulator);
+}).ToList();
 
-Task.WaitAll(serverTasks.ToArray());
-s1.Stop();
-Console.WriteLine($"creating 90 servers in time: {s1.ElapsedMilliseconds} ms");
+var matchRequestRepository = new CassandraMatchRequestRepository(mapper, ConsistencyLevel.One);
+var priorityManager = new PriorityManager(matchRequestRepository);
+var priorityManagerThread = new Thread(new ThreadStart(priorityManager.PriorityManagerLoop));
+priorityManagerThread.Start();
```

Comment:
Jakby program nie chciał się zakończyć, to można usunąć tą linijkę (i joina). Na większą skalę ten PriorityManager powinien działać, ale na to co mamy tutaj wpływa raczej negatywnie (a przynajmniej na moim laptopie)